src/llmcompressor/logger.py

```diff
@@ -93,30 +97,47 @@ def configure_logger(config: Optional[LoggerConfig] = None) -> None:
             sys.stdout,
             level=logger_config.console_log_level.upper(),
             format="{time} | {function} | {level} - {message}",
+            filter=support_log_once,
         )
 
     if logger_config.log_file or logger_config.log_file_level:
         log_file = logger_config.log_file or "llmcompressor.log"
         log_file_level = logger_config.log_file_level or "INFO"
         # log as json to the file for easier parsing
-        logger.add(log_file, level=log_file_level.upper(), serialize=True)
+        logger.add(
+            log_file,
+            level=log_file_level.upper(),
+            serialize=True,
+            filter=support_log_once,
+        )
 
     if logger_config.metrics_disabled or "METRIC" in logger._core.levels.keys():
         return
 
     # initialize metric logger on loguru
-    _initialize_metric_logging()
+    logger.level("METRIC", no=38, color="<yellow>", icon="📈")
 
 
-def _initialize_metric_logging() -> None:
+def support_log_once(record: Dict[str, Any]) -> bool:
     """
-    Initialize metric logging for loguru and metric-related libraries
-    Defaults to stdout
-    usage:
-        `logger.log("METRIC", "foo description, bar result")`
+    Support logging only once using `.bind(log_once=True)`
 
+    ```
+    logger.bind(log_once=True).info("This will only log once")
+    logger.bind(log_once=True).info("This will only log once")
```

Comment:
redundant line, maybe you meant to show the converse?
```suggestion
    logger.bind(log_once=False).info("This will log multiple times")
```

Reply:
Not really, I've updated with more exampling